internal/creds/processCreds.go

```diff
@@ -0,0 +1,34 @@
+package creds
+
+import (
+	"errors"
+	"log/slog"
+)
+
+type Auth struct {
+	Username string `yaml:"username"`
+	Password string `yaml:"password"`
+}
+
+type CredentialsList struct {
+	Users []Auth `yaml:"auth"`
+}
+
+var CredsFromFile CredentialsList
+
+func GetCredentials() (CredentialsList, error) {
+	if len(CredsFromFile.Users) == 0 {
+		slog.Error("CredentialsList is empty", "error", errors.New("Credentials are empty."))
+		return CredentialsList{}, errors.New("Credentials are empty.")
```

Comment:
If the list is empty this can be the signal to disable basic auth. It looks like returning an error here is causing the [linearizability check](https://github.com/resonatehq/resonate/actions/runs/9741229922/job/26907803484#step:6:8) to fail

Reply:
Yes.